### ISSUE TITLE
atom startup reports "IDE-PHP requires PHP 7 or later but found"

### ISSUE THREAD

**Issue by bsussman** [atom/ide-php#49]
Not really a complete sentence.

ide-php v 0.6.8
atom v 1.21.0beta2

my local apache2 is set up to use php 5.6.30-0+deb8u1, reported by standard phpinfo. which may have some bearing on this issue.  That would be interesting since I didn't think that atom used local stack setup to run.

2 buttons are presented. - "Set PHP Path" does nothing.
As this console log snip shows, php 7 or later is indeed installed:

```
[07:08] ~$-->  which php
/usr/bin/php
[07:45] ~$-->  php --version
PHP 7.1.9-1+0~20170902060604.8+jessie~1.gbpebe5d6 (cli) (built: Sep  2 2017 06:21:43) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.9-1+0~20170902060604.8+jessie~1.gbpebe5d6, Copyright (c) 1999-2017, by Zend Technologies
[07:45] ~$-->  
```
In fact, the settings shows this to be the executable already set:

![untitled](https://user-images.githubusercontent.com/87822/30912168-89df4480-a359-11e7-9531-f3f52c73a51f.png)

**Comment by lachieh** [comment]
I have this issues as well.

ide-php 0.6.8
atom 1.21.0beta2
macOS 10.12.6

I installed php with homebrew, so I tried setting the PHP Path in the ide-php config to the direct file `/usr/local/Cellar/php71/7.1.8_20/bin/php` rather than the symlink at `/usr/local/bin/php` but that didn't work either.

**Comment by lachieh** [comment]
On further investigation, it seems that there are two things happening.

The first is that because `stdOut` is [set to an empty string](https://github.com/atom/ide-php/blob/master/lib/main.js#L65), instead of being left undefined or set specifically to `null`, the [test on L69](https://github.com/atom/ide-php/blob/master/lib/main.js#L69) does not do as expected because `'' != null` returns true. This is a quick fix:

```diff
-    let stdErr = '', stdOut = ''
+    let stdErr, stdOut
```
This will leave both variables as available but the value as `undefined` and this works because `undefined != null` returns false

The second issue is that the [promise](https://github.com/atom/ide-php/blob/master/lib/main.js#L67) for `stdout` on L67 resolves after the process exits which means that at the time the exit function runs, the `stdOut` variable is still set to it's initial value of an empty string.
```js
childProcess.stdout.on('data', chunk => stdOut += chunk.toString())
```
I'm not too familiar with `child_process` but it would seem the `checkPHPVersion` method would need to be refactored so that it uses the `spawnSync` method instead of the async `spawn` method.

**Comment by lachieh** [comment]
aaaand there's already a PR open for to fix this: #45
